doc/source/serve/production.md

```diff
@@ -583,3 +565,218 @@ This section summarizes the best practices when deploying to production:
 * Use `serve config` to check the latest config that your Serve application received. This is its goal state.
 * Make lightweight configuration updates (e.g. `num_replicas` or `user_config` changes) by modifying your Serve config file and redeploying it with `serve deploy`.
 * Make heavyweight code updates (e.g. `runtime_env` changes) by starting a new Ray cluster, updating your Serve config file, and deploying the file with `serve deploy` to the new cluster. Once the new deployment is finished, switch your traffic to the new cluster.
+
+(serve-in-production-kubernetes)=
+
+## Deploying to Production on Kubernetes
+
+The above sections walk through how to deploy a Serve application manually to a running Ray cluster using the Serve CLI.
```

Comment:
I think it is a bit odd that we start off saying that we recommend using Kubernetes to deploy, but then we jump immediately to walking through the not-so-recommended way. The page is also getting a bit long. Maybe we can:

1. Split out the Kubernetes part into its own page.
2. Make it clear that we will use the CLI/local dev example as a quickstart, but guide the user to the Kubernetes guide for actual production at the beginning/end.